src/pyosmeta/file_io.py

```diff
@@ -21,124 +20,158 @@ def get_api_token() -> str:
     return os.environ["GITHUB_TOKEN"]
 
 
-@dataclass
-class YamlIO:
+def load_pickle(filename):
+    """Opens a pickle"""
+    with open(filename, "rb") as f:
+        return pickle.load(f)
+
+
+def _list_to_dict(a_list: list, a_key: str) -> dict:
+    """Takes a yaml file opened and turns into a dictionary
+    The dict structure is key (gh_username) and then a dictionary
+    containing all information for the username
+
+    a_list : list
+        A list of dictionary objects returned from load website yaml
+    a_key : str
+        A string representing the dict key to use as the main key to call
+        each sub dict in the object.
+
+    """
+    final_dict = {}
+    for dict in a_list:
+        # All github usernames are lower to make this a key
+        final_dict[dict[a_key].lower()] = dict
+
+    return final_dict
+
+
+def load_website_yml(a_key: str, a_url: str):
     """
-    A class that provides file output and cleanup methods.
+    This opens a website contrib yaml file and turns it in a
+    dictionary
+    """
+    yml_list = open_yml_file(a_url)
+
+    return _list_to_dict(yml_list, a_key)
+
+
+def dict_to_list(pyosDict: dict) -> list:
```

Comment:
instad of `pyosDict` (cammel case), consider using snake case. also maybe it would be nice something more descriptive, but maybe it is a very generic dictionary, so maybe you can use something like `metadata`, `github_data`, etc. Normally it is not a good approach to have the type of object as part of the variable name ... but I guess that if you have the same kind of data in a different type, for example a date in string format and the same date in datetime type, it makes sense to me to have the type as part of the variable name, but I guess that in this case it is not necessary.

Reply:
also, maybe you could improve the type annotation to add more information, for example,
instead of `list`, you could have `List[str]` to represent a list of strings.

but, maybe this function is not super necessary as python already have a built-in function to get the value from a dictionary:

```
>>> {"a": 1, "b": 2, "c": 3}.values()
dict_values([1, 2, 3])
```

Reply:
i've fixed most of this but for some reason when i call `.values()` on the Dict i get a representation error when i try to export my data to YAML. i still haven't figured out why.

Reply:
chatgpt for the win

```
>>> list({"a": 1, "b": 2, "c": 3}.values())
[1, 2, 3]
```

if you don't make it a list then it just returns a dict_values() thing that is not a list!

Reply:
some of this stuff i just haven't had time to dig into and refine. but i still have design questions generally. but will make these fixes for now as it's cleaner!!